plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxApiClient.java

```diff
@@ -453,8 +453,9 @@ protected void removeSegment(String segmentName) {
         String siteId = getDefaultSiteId();
         String enforcementPointPath = getDefaultEnforcementPointPath(siteId);
         SegmentPorts segmentPortsService = (SegmentPorts) nsxService.apply(SegmentPorts.class);
-        PolicyGroupMembersListResult segmentPortsList = segmentPortsService.list(DEFAULT_DOMAIN, segmentName, null, enforcementPointPath,
-                false, null, 50L, false, null);
+        PolicyGroupMembersListResult segmentPortsList = getSegmentPortList(segmentPortsService, segmentName, enforcementPointPath);
+        Long portCount = segmentPortsList.getResultCount();
+        portCount = retrySegmentDeletion(segmentPortsService, portCount, segmentName, enforcementPointPath);
```

Comment:
I think you can just leave this line defining Long portCount and remove the previous 2 lines